app/celery/contact_information_tasks.py

```diff
@@ -1,18 +1,66 @@
+from typing import Optional
+from sqlalchemy.orm.exc import NoResultFound
 from flask import current_app
 from app import notify_celery, va_profile_client
 from app.celery.common import can_retry, handle_max_retries_exceeded
 from app.celery.exceptions import AutoRetryException
 from app.celery.service_callback_tasks import check_and_queue_callback_task
+from app.feature_flags import FeatureFlag, is_feature_enabled
 from app.va.identifier import IdentifierType
 from app.va.va_profile import VAProfileRetryableException, VAProfileNonRetryableException, NoContactInfoException
+from app.dao.communication_item_dao import get_communication_item
 from app.dao.notifications_dao import get_notification_by_id, dao_update_notification, update_notification_status_by_id
-from app.models import NOTIFICATION_PERMANENT_FAILURE, EMAIL_TYPE, SMS_TYPE
+from app.models import NOTIFICATION_PERMANENT_FAILURE, NOTIFICATION_PREFERENCES_DECLINED, EMAIL_TYPE, SMS_TYPE
 from app.exceptions import NotificationTechnicalFailureException, NotificationPermanentFailureException
-from app.va.va_profile.exceptions import VAProfileIDNotFoundException
+from app.va.va_profile.exceptions import (
+    VAProfileIDNotFoundException,
+    CommunicationItemNotFoundException,
+    CommunicationPermissionDenied,
+)
 from notifications_utils.statsd_decorators import statsd
 from requests import Timeout
 
 
+def get_recipient(notification_type, notification_id, recipient_identifier, communication_item_id_for_permission_check):
+    if notification_type == EMAIL_TYPE:
+        return get_email_recipient(recipient_identifier, communication_item_id_for_permission_check)
+    elif notification_type == SMS_TYPE:
+        return get_sms_recipient(recipient_identifier, communication_item_id_for_permission_check)
+    else:
+        raise NotImplementedError(
+            f'The task lookup_contact_info failed for notification {notification_id}. '
+            f'{notification_type} is not supported'
+        )
```

Comment:
For new code we would like to see type annotations for parameters and return types. It's also a general rule to only have one return from a function.
i.e.
```python
def make_str(arg1: str, arg2: int) -> str:
    return f'{arg1} is {arg2}'
```

Reply:
I just added type annotations and doc strings.  As for the multiple return statements, I don't see an easy way to avoid it.  This should be temporary anyway.  The `get_recipient` function was added because the linter was not happy about deeply nested if statements, and can be removed once we take out the `VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP` feature flag.